lib/database/set-claim.php

```diff
@@ -0,0 +1,500 @@
+        if ($getCount) {
+            return 0;
+        }
+        return $retVal;
+    }
+
+    // Create set type condition
+    $setTypeCondition = '';
+    if ($newSetClaim && !$revisionClaim) {
+        $setTypeCondition = 'AND sc.SetType = 0';
+    } elseif (!$newSetClaim && $revisionClaim) {
+        $setTypeCondition = 'AND sc.SetType = 1';
+    } elseif (!$newSetClaim && !$revisionClaim) {
+        if ($getCount) {
+            return 0;
+        }
+        return $retVal;
+    }
+
+    // Create the claim status condition
+    $statusCondition = '';
+    if ($activeClaim && $completeClaim && !$droppedClaim) {
+        $statusCondition = 'AND sc.Status IN (0, 1)';
+    } elseif ($activeClaim && !$completeClaim && $droppedClaim) {
+        $statusCondition = 'AND sc.Status IN (0, 2)';
+    } elseif ($activeClaim && !$completeClaim && !$droppedClaim) {
+        $statusCondition = 'AND sc.Status = 0';
+    } elseif (!$activeClaim && $completeClaim && $droppedClaim) {
+        $statusCondition = 'AND sc.Status IN (1, 2)';
+    } elseif (!$activeClaim && $completeClaim && !$droppedClaim) {
+        $statusCondition = 'AND sc.Status = 1';
+    } elseif (!$activeClaim && !$completeClaim && $droppedClaim) {
+        $statusCondition = 'AND sc.Status = 2';
+    } elseif (!$activeClaim && !$completeClaim && !$droppedClaim) {
+        if ($getCount) {
+            return 0;
+        }
+        return $retVal;
+    }
+
+    // Create the developer status condition
+    $devStatusCondition = '';
+    if ($developerClaim && !$juniorDeveloperClaim) {
+        $devStatusCondition = "AND ua.Permissions >= " . Permissions::Developer;
+    } elseif (!$developerClaim && $juniorDeveloperClaim) {
+        $devStatusCondition = "AND ua.Permissions = " . Permissions::JuniorDeveloper;
+    } elseif (!$developerClaim && !$juniorDeveloperClaim) {
+        $devStatusCondition = "AND ua.Permissions < " . Permissions::JuniorDeveloper;
+    }
+
+    // Determine ascending or descending order
+    if ($sortType < 10) {
+        $sortOrder = "DESC";
+    } else {
+        $sortOrder = "ASC";
+        $sortType = $sortType - 10;
+    }
+
+    // Create the sorting condition
+    $sortCondition = match ($sortType) {
+        2 => 'sc.User ',
+        3 => 'gd.Title ',
+        4 => 'sc.ClaimType ',
+        5 => 'sc.SetType ',
+        6 => 'sc.Status ',
+        7 => 'sc.Special ',
+        8 => 'sc.Created ',
+        9 => 'sc.Finished ',
+        default => 'sc.Created ',
+    }
+    . $sortOrder;
+
+    // Creare the user data condition
+    $userCondition = '';
+    if (isset($username)) {
+        $userCondition = "AND sc.User = '$username'";
+    }
+
+    // Create the game condition
+    $gameCondition = '';
+    if ($gameID > 0) {
+        $gameCondition = "AND sc.GameID = '$gameID'";
+    }
+
+    // Get expiring claims only
+    $havingCondition = '';
+    if ($getExpiringOnly) {
+        $havingCondition = "HAVING MinutesLeft <= 10080";
```

Comment:
recommend adding a comment for deriving this value: `// 7 days = 7 * 24 * 60`

Reply:
Added suggested comments.